include/property.h

```diff
@@ -154,6 +156,15 @@ void parse_props(LogRecord &p, std::string &line,
 std::ostream &operator<<(std::ostream &ss, const LogRecord &lr) {
   return util::to_string(ss, lr.tie());
 }
+
+static std::string StateStart("");
+static std::string StateSep(" ; ");
+static std::string StateEnd("");
+std::ostream &operator<<(std::ostream &s,
+                          const std::pair<CodeContext, RunContext> &p){
+  return s << StateStart << p.first << StateSep << p.second
+           << StateEnd;
+}
```

Comment:
can we use util::to_string? pair is a tuple, right? if not, can we use 2-tuple instead of pair?